### PR TITLE
Adds Hereditary Manifold Sickness, a Chronic Illness Quirk

### PR DESCRIPTION
**Pull request by orthography**
## About The Pull Request

Adds a new quirk called Chronic Illness. It provides a -12 score as it's a pretty life-altering quirk, but could give way to interesting RP.

- Hereditary Manifold Sickness (HMS) can not be cured, it can only be delayed and treated using a new unmakable vaccine called "Sansufentanyl". You spawn with 6 pills and are able to order crates containing 12 more from cargo as it's proprietary to Interdyne.

- HMS has 5 stages total.
Stage 1 does nothing
Stage 2 gives minor effects 
Stage 3 becomes debilitating
Stage 4 is a danger zone.

Upon reaching Stage 5, there are 4 possibilities, 1 is a recovery back to stage 1, and the other 3 are deaths which I won't explain here to avoid ruining it. (read the code I guess.)

This also adds a new traitor objective to infect someone with HMS.

- [x] Correcting the chronic aspect.

- [x] Add traitor objective.

- [x] Tweaking for fairness.

## Why It's Good For The Game

HMS is a new quirk that gives a much more hardcore junky or tumor playstyle. Neglecting HMS can mean the end of your shift. it's not something you want to mess with.

It puts a reliance on cargo rather than medical for a quirk and gives a use case for money. (price may be tweaked still). I think it'd be interesting to see if people will start mugging or robbing the vault more to get their life-saving medication.

## Changelog
:cl:
add: Interdyne has released a new medication to treat those who are in the wrong timeline!
add: Interdyne has also realized this is VERY profitable! They've begun arming their operatives with an autoinjector.
/:cl:

### CODE REVIEW

File: code/__DEFINES/traits.dm [hunk 1]
```diff
@@ -628,6 +628,7 @@ Remember to update _globalvars/traits.dm if you're adding/removing/renaming trai
 #define TRAIT_NIGHT_VISION "night_vision"
 #define TRAIT_LIGHT_STEP "light_step"
 #define TRAIT_SPIRITUAL "spiritual"
+#define TRAIT_CHRONIC_ILLNESS "chronic_illness"
```
Comment (Fikou) [inline code/__DEFINES/traits.dm:631]: why did you add this if it does nothing

File: code/datums/diseases/_disease.dm [hunk 2]
```diff
@@ -67,12 +67,17 @@
...
 	if(has_cure())
-		if(SPT_PROB(cure_chance, seconds_per_tick))
-			update_stage(max(stage - 1, 1))
-
-		if(disease_flags & CURABLE && SPT_PROB(cure_chance, seconds_per_tick))
-			cure()
-			return FALSE
+		if(disease_flags & CHRONIC && SPT_PROB(cure_chance, seconds_per_tick))
+			src.stage = 1
+			to_chat(affected_mob, span_notice("You feel condition improve."))
```
Comment (tralezab) [inline code/datums/diseases/_disease.dm:72]: "You feel condition improve." should clean up this line of text

File: code/datums/diseases/_disease.dm [hunk 3]
```diff
@@ -1,7 +1,7 @@
 /datum/disease
 	//Flags
 	var/visibility_flags = 0
-	var/disease_flags = CURABLE|CAN_CARRY|CAN_RESIST
+	var/disease_flags = CURABLE|CAN_CARRY|CAN_RESIST|CHRONIC
```
Comment (Fikou) [inline code/datums/diseases/_disease.dm:4]: this cannot be a default flag. there are diseases that are subtypes of this that we don't want to randomly set to stage 1

**Comment by tralezab** [comment]
Super cool though!

File: code/datums/diseases/chronic_illness.dm [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+		if(4)
+			to_chat(affected_mob, span_warning("You feel yourself phasing in and out of reality! You need your medication!"))
+			affected_mob.playsound_local(affected_mob, 'sound/effects/singlebeat.ogg', 100, FALSE, use_reverb = FALSE)
+			if(SPT_PROB(1, seconds_per_tick))
+				to_chat(affected_mob, span_danger("You feel an grusome pain in your chest!"))
+				if(prob(75))
+					affected_mob.vomit(45,TRUE)
+			if(SPT_PROB(1, seconds_per_tick))
+				affected_mob.adjustStaminaLoss(100, FALSE)
+				affected_mob.visible_message(span_warning("[affected_mob] faints!"), span_userdanger("You feel your atoms accelerate it brings you to the floor."))
```
Comment (Fikou) [inline code/datums/diseases/chronic_illness.dm:60]: not good grammar

File: code/datums/diseases/chronic_illness.dm [hunk 5]
```diff
@@ -0,0 +1,104 @@
...
+					to_chat(affected_mob, span_boldwarning("There is no place for you in this timeline."))
+					affected_mob.adjustStaminaLoss(100, forced = TRUE)
+					playsound(affected_mob.loc, 'sound/magic/repulse.ogg', 100, FALSE)
+					affected_mob.emote("scream")
+					for(var/mob/living/L in viewers(3, affected_mob.loc))
+						L.flash_act()
+					New(affected_mob.loc, /obj/effect/decal/cleanable/plasma)
+					var/time_delta = world.time
+					if(world.time == time_delta + 2)
+						return
```
Comment (Fikou) [inline code/datums/diseases/chronic_illness.dm:89]: when would this be ever true?

File: code/datums/diseases/chronic_illness.dm [hunk 6]
```diff
@@ -0,0 +1,104 @@
...
+					to_chat(affected_mob, span_notice("You feel your atoms begin to realign. You're safe. For now."))
+					src.stage = 1
+				if(2)
+					to_chat(affected_mob, span_boldwarning("There is no place for you in this timeline."))
+					affected_mob.adjustStaminaLoss(100, forced = TRUE)
+					playsound(affected_mob.loc, 'sound/magic/repulse.ogg', 100, FALSE)
+					affected_mob.emote("scream")
+					for(var/mob/living/L in viewers(3, affected_mob.loc))
+						L.flash_act()
+					New(affected_mob.loc, /obj/effect/decal/cleanable/plasma)
```
Comment (Fikou) [inline code/datums/diseases/chronic_illness.dm:86]: this is not the correct way to create new atoms, do 
new /typepath(location)

File: code/datums/diseases/chronic_illness.dm [hunk 7]
```diff
@@ -0,0 +1,104 @@
...
+			switch(pick(1,2))
+				if(1)
+					to_chat(affected_mob, span_notice("You feel your atoms begin to realign. You're safe. For now."))
+					src.stage = 1
+				if(2)
+					to_chat(affected_mob, span_boldwarning("There is no place for you in this timeline."))
+					affected_mob.adjustStaminaLoss(100, forced = TRUE)
+					playsound(affected_mob.loc, 'sound/magic/repulse.ogg', 100, FALSE)
+					affected_mob.emote("scream")
+					for(var/mob/living/L in viewers(3, affected_mob.loc))
```
Comment (Fikou) [inline code/datums/diseases/chronic_illness.dm:84]: use descriptive variable names, not single letters

File: code/datums/quirks/negative_quirks.dm [hunk 8]
```diff
@@ -889,6 +889,29 @@
...
+/datum/quirk/item_quirk/chronic_illness
+	name = "Chronic Illness"
+	desc = "You have a chronic illness that requires constant medication to keep under control."
+	icon = FA_ICON_DISEASE
+	value = -8
+	mob_trait = TRAIT_CHRONIC_ILLNESS
+	gain_text = span_danger("You feel a bit off today.")
+	lose_text = span_notice("You feel a bit better today.")
+	medical_record_text = "Patient has a chronic illness that requires constant medication to keep under control."
+	hardcore_value = 6
```
Comment (Time-Green) [inline code/datums/quirks/negative_quirks.dm:901]: Only six is pretty rough, since it can randomly decide your new objective is to just make money to barely not die. Paraplegic and blind are 15, for reference. I think something like 10 would be more fair

File: code/datums/diseases/chronic_illness.dm [hunk 9]
```diff
@@ -0,0 +1,104 @@
...
+			if(SPT_PROB(1, seconds_per_tick))
+				switch(rand(1,3))
+					if(1)
+						scramble_dna(affected_mob, 1, 0, 0, rand(15,45))
+					if(2)
+						scramble_dna(affected_mob, 0, 1, 0, rand(15,45))
+					if(3)
+						scramble_dna(affected_mob, 0, 0, 1, rand(15,45))
+		if(5)
+			switch(pick(1,2))
```
Comment (Time-Green) [inline code/datums/diseases/chronic_illness.dm:75]: So you can't roll 3 and 4?

**Comment by Time-Green** [comment]
How long does it take to progress from stage 1 to stage 5?

File: code/__DEFINES/traits.dm (hunk 1 above)
Comment (orthography) [reply code/__DEFINES/traits.dm:631]: Trying to find a way to fix bug 1 I mentioned in the PR.

File: code/datums/diseases/chronic_illness.dm (hunk 9 above)
Comment (orthography) [reply code/datums/diseases/chronic_illness.dm:75]: That's just for testing right now.

**Comment by orthography** [comment]
> How long does it take to progress from stage 1 to stage 5?

It's randomized. The way diseases work you have a 0.25% chance for it to advance every second.

File: code/datums/quirks/negative_quirks.dm (hunk 8 above)
Comment (orthography) [reply code/datums/quirks/negative_quirks.dm:901]: Agreed

File: code/__DEFINES/diseases.dm [hunk 10]
```diff
@@ -8,6 +8,7 @@
 
 //Disease Flags
 #define CURABLE (1<<0)
+#define CHRONIC (1<<0)
```
Comment (NamelessFairy) [inline code/__DEFINES/diseases.dm:11]: Curable and Chronic appear to be using the same value, I'm pretty sure this is going to cause them to be treated as the same when used later. You probably want to give chronic a new value. (1<<3) should be fine probably.

**Comment by KittyNoodle** [comment]
assistant arrives on station, manages to roll 0.25% chance 5 times in a row, dies instantly. best quirk ever

**Comment by orthography** [comment]
> assistant arrives on station, manages to roll 0.25% chance 5 times in a row, dies instantly. best quirk ever

Chance of this happening is 9.765625×10−14 if my math isn't ass. If you somehow get this and hit that lottery, sucks to be you, you silly little non-zero possibility!

**Comment by orthography** [comment]
@Fikou Can you re-review after the checks?

**Comment by Wallemations** [comment]
I'll be honest, this feels pretty boring for another negative trait, we already have a couple "take chem regularly or bad thing happens" traits. 
If you could manage to make it something else instead of a chemical then that might be more interesting, but unless you were having to hunt down dimensional tears to harvest their energy to keep yourself in the timeline, I doubt you could really do anything.

File: code/datums/diseases/chronic_illness.dm [hunk 11]
```diff
@@ -0,0 +1,103 @@
...
+	spread_flags = DISEASE_SPREAD_NON_CONTAGIOUS
+	disease_flags = CHRONIC
+	process_dead = TRUE
+	stage_prob = 0.25
+	cure_text = "Sansufentanyl"
+	cures = list(/datum/reagent/medicine/sansufentanyl)
+	infectivity = 0
+	agent = "Quantum Entanglement"
+	viable_mobtypes = list(/mob/living/carbon/human)
+	desc = "A disease discovered in an Interdyne laboratory caused by the subjection to timesteam correction technology."
```
Comment (ATH1909) [inline code/datums/diseases/chronic_illness.dm:14]:
```suggestion
	desc = "A disease discovered in an Interdyne laboratory caused by subjection to timesteam correction technology."
```

File: code/datums/diseases/chronic_illness.dm [hunk 12]
```diff
@@ -0,0 +1,103 @@
...
+		if(3)
+			if(SPT_PROB(0.5, seconds_per_tick))
+				to_chat(affected_mob, span_danger("You feel a very sharp pain in your chest!"))
+				if(prob(45))
+					affected_mob.vomit(20)
+			if(SPT_PROB(0.5, seconds_per_tick))
+				to_chat(affected_mob, span_userdanger("[pick("You feel your heart slowing...", "You relax and slow your heartbeat.")]"))
+				affected_mob.adjustStaminaLoss(70, FALSE)
+			if(SPT_PROB(1, seconds_per_tick))
+				to_chat(affected_mob, span_danger("You feel a buzzing in the brain."))
```
Comment (ATH1909) [inline code/datums/diseases/chronic_illness.dm:39]:
```suggestion
				to_chat(affected_mob, span_danger("You feel a buzzing in your brain."))
```

File: code/datums/diseases/chronic_illness.dm [hunk 13]
```diff
@@ -0,0 +1,103 @@
...
+				if(3)
+					affected_mob.visible_message(span_warning("[affected_mob] is torn apart!"), span_userdanger("Your atoms accelerate into criticallity!"))
+					affected_mob.gib()
+				if(4)
+					if(affected_mob.stat == CONSCIOUS)
+						affected_mob.visible_message(span_danger("[affected_mob] clutches at [affected_mob.p_their()] chest as if [affected_mob.p_their()] heart is stopping!"), \
+						span_userdanger("You feel a horrible pain as your heart phases out of your chest!"))
+						var/obj/item/organ/internal/heart/heart = affected_mob.get_organ_slot(ORGAN_SLOT_HEART)
+						qdel(heart)
+						src.stage = 1
```
Comment (ATH1909) [inline code/datums/diseases/chronic_illness.dm:100]: is it intentional that you can dodge the heart ripping/stage resetting by being unconscious

File: code/datums/quirks/negative_quirks.dm [hunk 14]
```diff
@@ -889,6 +889,28 @@
 			return
 		quirk_holder.add_mood_event("wrong_cigs", /datum/mood_event/wrong_brand)
 
+/datum/quirk/item_quirk/chronic_illness
+	name = "Chronic Illness"
+	desc = "You have a chronic illness that requires constant medication to keep under control."
+	icon = FA_ICON_DISEASE
+	value = 0
```
Comment (ATH1909) [inline code/datums/quirks/negative_quirks.dm:896]: update the opening post if this does not actually give 8 points

File: code/datums/diseases/chronic_illness.dm [hunk 15]
```diff
@@ -0,0 +1,103 @@
...
+	spread_text = "Unspread Illness"
+	spread_flags = DISEASE_SPREAD_NON_CONTAGIOUS
+	disease_flags = CHRONIC
+	process_dead = TRUE
+	stage_prob = 0.25
+	cure_text = "Sansufentanyl"
+	cures = list(/datum/reagent/medicine/sansufentanyl)
+	infectivity = 0
+	agent = "Quantum Entanglement"
+	viable_mobtypes = list(/mob/living/carbon/human)
```
Comment (ATH1909) [inline code/datums/diseases/chronic_illness.dm:13]: IIRC, you might also need to expand the infectable_biotypes, although I could be wrong here

might also need bypasses_immunity = TRUE?

**Comment by ATH1909** [comment]
you should probably also give them some way to check of the progress of their illness. either, like, a specialized health analyzer in their starting kit or some guaranteed message that they're given on each stage transition.

otherwise, you're kind of forced to rush medbay or robotics at roundstart for a health analyzer.

File: code/datums/diseases/chronic_illness.dm [hunk 16]
```diff
@@ -0,0 +1,103 @@
...
+				to_chat(affected_mob, span_warning("You feel dizzy."))
+				affected_mob.adjust_confusion(6 SECONDS)
+			if(SPT_PROB(0.5, seconds_per_tick))
+				to_chat(affected_mob, span_notice("You look at your hand. Your vision blurs."))
+				affected_mob.set_eye_blur_if_lower(10 SECONDS)
+		if(3)
+			if(SPT_PROB(0.5, seconds_per_tick))
+				to_chat(affected_mob, span_danger("You feel a very sharp pain in your chest!"))
+				if(prob(45))
+					affected_mob.vomit(20)
```
Comment (ATH1909) [inline code/datums/diseases/chronic_illness.dm:34]: friendly reminder that vomiting is a lengthy hardstun

this probably shouldn't be on stage 3

File: code/datums/diseases/chronic_illness.dm (hunk 15 above)
Comment (orthography) [reply code/datums/diseases/chronic_illness.dm:13]: It's non_contagious, I'll check it out though. It shouldn't need bypass immunity because it's incurable. I'll write it up though.

**Comment by ophaq** [comment]
If you are adding this, please add this "Sansufentanyl" to the list of possible chems to be allergic to for added comedic effect.

**Comment by orthography** [comment]
> I'll be honest, this feels pretty boring for another negative trait, we already have a couple "take chem regularly or bad thing happens" traits. If you could manage to make it something else instead of a chemical then that might be more interesting, but unless you were having to hunt down dimensional tears to harvest their energy to keep yourself in the timeline, I doubt you could really do anything.

I worry about doing something like this because it throws off the entire gameplay loop. I don't want this quirk to turn into; running around the station as a shittier heretic to avoid death, instead of just doing your job with constant inconveniences.

When I played a few rounds, I quickly felt overwhelmed by people trying to steal my meds, or losing them in a fight and being overwhelmed to recover it, it definitely added to the story, 

I recognize there are a lot of med quirks, and I'll entertain options, but I think this one is a bit more overwhelming as a junkie isn't a requirement, brain tumor is recoverable, and this can RR you if you're not careful.

**Comment by Loafin34** [comment]
really annoying idea incoming: what if this was a traitor objective. "infect [name] with the experimental bluespace sickness" and it gave you some one-use autoinjector for it. itd make people more paranoid of tiny pricks, and the thingll give you plenty warning of "oh shit i got manifold'd"

**Comment by CMDR-Gungnir** [comment]
> really annoying idea incoming: what if this was a traitor objective. "infect [name] with the experimental bluespace sickness" and it gave you some one-use autoinjector for it. itd make people more paranoid of tiny pricks, and the thingll give you plenty warning of "oh shit i got manifold'd"

I kinda like this idea, and not just as a replacement. I think it'd be pretty Neat alongside it.

**Comment by optimumtact** [comment]
Error: /datum/traitor_objective/target_player/infect is in a traitor category and is an abstract type! Please remove it from the /datum/traitor_objective_category.
  Error: /datum/traitor_objective/target_player/infect has set a progression reward as an abstract type! Please define progression rewards on non-abstract types rather than abstract types.
  Error: /datum/traitor_objective/target_player/infect has set a telecrystal reward as an abstract type! Please define telecrystal rewards on non-abstract types rather than abstract types.
  	FAILURE #1: /datum/traitor_objective/target_player/infect is in a traitor category and is an abstract type! Please remove it from the /datum/traitor_objective_category. at code/modules/unit_tests/objectives.dm:19
  	FAILURE #2: /datum/traitor_objective/target_player/infect has set a progression reward as an abstract type! Please define progression rewards on non-abstract types rather than abstract types. at code/modules/unit_tests/objectives.dm:22
  	FAILURE #3: /datum/traitor_objective/target_player/infect has set a telecrystal reward as an abstract type! Please define telecrystal rewards on non-abstract types rather than abstract types. at code/modules/unit_tests/objectives.dm:25

**Comment by orthography** [comment]
Woah, there is a god it passed.

This is ready.

File: code/datums/diseases/chronic_illness.dm [hunk 17]
```diff
@@ -0,0 +1,103 @@
...
+			if(SPT_PROB(0.5, seconds_per_tick))
+				affected_mob.adjustBruteLoss(1)
+			if(SPT_PROB(0.5, seconds_per_tick))
+				switch(rand(1,3))
+					if(1)
+						scramble_dna(affected_mob, 1, 0, 0, rand(15,45))
+					if(2)
+						scramble_dna(affected_mob, 0, 1, 0, rand(15,45))
+					if(3)
+						scramble_dna(affected_mob, 0, 0, 1, rand(15,45))
```
Comment (ChungusGamer666) [inline code/datums/diseases/chronic_illness.dm:52]: idk if it's just me but boy this needs named arguments

File: code/datums/diseases/chronic_illness.dm [hunk 18]
```diff
@@ -0,0 +1,103 @@
...
+			if(SPT_PROB(1, seconds_per_tick))
+				switch(rand(1,3))
+					if(1)
+						scramble_dna(affected_mob, 1, 0, 0, rand(15,45))
+					if(2)
+						scramble_dna(affected_mob, 0, 1, 0, rand(15,45))
+					if(3)
+						scramble_dna(affected_mob, 0, 0, 1, rand(15,45))
+		if(5)
+			switch(pick(1,2,3,4))
```
Comment (ChungusGamer666) [inline code/datums/diseases/chronic_illness.dm:79]: no reason not to use rand() here

**Comment by orthography** [comment]
Code maintainer Gods. Please send me a sign.

**Comment by nukularpower** [comment]
Love the idea here, hopefully it can be curable with a % chance by removing tumor'd limbs/organs or something (ie not gauranteed to cure, but some chance)

Also hope its possible to cause it with radiation.   Rads are embarassingly weak on TG now, and lets face it, in real life you do not recover from rad poisoning, no matter how much Potass Iodine you drink.  Would work nicely for sometime like this 

Anyways, love it

**Comment by Loafin34** [comment]
@nukularpower rads are a funny idea, but I'm a bit worried of a guy with HMS just working as an engineer to have free -12 quirk points.

**Comment by carshalash** [comment]
This seems... astoundingly common of an objective.

**Comment by orthography** [comment]
> This seems... astoundingly common of an objective.

See new PR. :)

**Comment by YehnBeep** [comment]
With this being forced onto people, I feel like it shouldn't bottleneck quite so hard on cargo.
